PhysicsTools/NanoAOD/python/nano_cff.py

```diff
@@ -304,59 +252,24 @@ def nanoAOD_runMETfixEE2017(process,isData):
 def nanoAOD_customizeCommon(process):
 
     process = nanoAOD_activateVID(process)
-    nanoAOD_addDeepInfo_switch = cms.PSet(
-        nanoAOD_addDeepBTag_switch = cms.untracked.bool(False),
-        nanoAOD_addDeepFlavourTag_switch = cms.untracked.bool(False),
-        )
-    run2_miniAOD_80XLegacy.toModify(nanoAOD_addDeepInfo_switch, nanoAOD_addDeepBTag_switch = cms.untracked.bool(True))
-    for modifier in run2_miniAOD_80XLegacy, run2_nanoAOD_94X2016, run2_nanoAOD_94XMiniAODv1, run2_nanoAOD_94XMiniAODv2:
-        modifier.toModify(nanoAOD_addDeepInfo_switch, nanoAOD_addDeepFlavourTag_switch =  cms.untracked.bool(True))
-    process = nanoAOD_addDeepInfo(process,
-                                  addDeepBTag=nanoAOD_addDeepInfo_switch.nanoAOD_addDeepBTag_switch,
-                                  addDeepFlavour=nanoAOD_addDeepInfo_switch.nanoAOD_addDeepFlavourTag_switch)
-    nanoAOD_addDeepInfoAK8_switch = cms.PSet(
-        nanoAOD_addDeepBTag_switch = cms.untracked.bool(False),
-        nanoAOD_addDeepBoostedJet_switch = cms.untracked.bool(False),
-        nanoAOD_addDeepDoubleX_switch = cms.untracked.bool(False),
-        nanoAOD_addDeepDoubleXV2_switch = cms.untracked.bool(False),
-        nanoAOD_addParticleNet_switch = cms.untracked.bool(False),
-        nanoAOD_addParticleNetMass_switch = cms.untracked.bool(False),
-        jecPayload = cms.untracked.string('AK8PFPuppi')
-        )
-    # deepAK8 should not run on 80X, that contains ak8PFJetsCHS jets
-    run2_miniAOD_80XLegacy.toModify(nanoAOD_addDeepInfoAK8_switch,
-                                    nanoAOD_addDeepBTag_switch = True,
-                                    jecPayload = 'AK8PFchs')
-    # for 94X and 102X samples: needs to run DeepAK8, DeepDoubleX and ParticleNet
-    (run2_nanoAOD_94X2016 | run2_nanoAOD_94XMiniAODv1 | run2_nanoAOD_94XMiniAODv2 | run2_nanoAOD_102Xv1).toModify(
-        nanoAOD_addDeepInfoAK8_switch,
-        nanoAOD_addDeepBoostedJet_switch = True,
-        nanoAOD_addDeepDoubleX_switch = True,
-        nanoAOD_addDeepDoubleXV2_switch = True,
-        nanoAOD_addParticleNet_switch = True,
-        nanoAOD_addParticleNetMass_switch = True,
-    )
-    # for 106Xv1: only needs to run ParticleNet and DDXV2; DeepAK8, DeepDoubleX are already in MiniAOD
-    run2_nanoAOD_106Xv1.toModify(
-        nanoAOD_addDeepInfoAK8_switch,
-        nanoAOD_addDeepDoubleXV2_switch = True,
-        nanoAOD_addParticleNet_switch = True,
-        nanoAOD_addParticleNetMass_switch = True,
-    )
 
-    run2_nanoAOD_106Xv2.toModify(
-         nanoAOD_addDeepInfoAK8_switch,
-         nanoAOD_addParticleNetMass_switch = True,
+    # This function is defined in jetsAK4_CHS_cff.py
+    process = nanoAOD_addDeepInfoAK4CHS(process,
```

Comment:
Yes. They are only needed for Run-2 since the AK4 CHS jets will only be for Run 2 NanoAODs and for the AK8 Puppi jets, all the discriminators should be available in the input Run 3 MiniAODs.

Reply:
can you put these beyond the run2_nanoAOD_ANY? so that will be clear when reading

Reply:
@mariadalfonso  Apologies. I may have poorly explained this.  These two functions rely on the switches defined [here (for nanoAOD_addDeepInfoAK8)](https://github.com/nurfikri89/cmssw/blob/f3d6e210e94fa48d8f497fd648d20cc5895e5896/PhysicsTools/NanoAOD/python/jetsAK8_cff.py#L228-L262) and [here for (nanoAOD_addDeepInfoAK4CHS)](https://github.com/nurfikri89/cmssw/blob/from1240pre3_nano_jetsrun3_forPR/PhysicsTools/NanoAOD/python/jetsAK4_CHS_cff.py#L362-L368). As you can see, the configuration for the switches depend on specific Run 2 era modifiers  rather the overall `run2_nanoAOD_ANY` modifier. Is your suggestion that, for readability,  the configurations for these switches be moved to just above where the two functions are called in `nano_cff.py`?

Reply:
we can simplify all this later.